utils/src/config.rs

```diff
@@ -5,14 +5,26 @@ use figment::{
     providers::{Env, Format, Serialized, Toml},
     Figment,
 };
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
 
-/// Trait with the basic to load configurations for different services.
+/// Environment variable default prefix.
+pub const ENV_PREFIX: &str = "MIDEN__";
+
+/// The `(host, port)` pair for the server's listening socket.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
+pub struct HostPort {
+    /// Host used by the store.
+    pub host: String,
+    /// Port number used by the store.
+    pub port: u16,
+}
```

Comment:
I find the `HostPort` naming a bit confusing (imo, just `Endpoint` should have been fine) - but we can find a better name in a subsequent PR.